lib/iot-core-jobs/source/core_jobs.c

```diff
@@ -55,37 +58,41 @@ static size_t getUpdateJobExecutionMsg( JobStatus_t status,
                                         size_t expectedVersionLength,
                                         char * buffer,
                                         size_t bufferSize );
+static bool isThingnameTopicMatch(const char * topic,
+                                    const size_t topicLength,
+                                    const char * topicSuffix,
+                                    const size_t topicSuffixLength );
 
 /* TODO: Consider creating a 'thingName topic matching' function. This could simply call that function */
 bool coreJobs_isStartNextAccepted( const char * topic,
                                    const size_t topicLength )
 {
-    /* TODO: Inefficient - better implementation shouldn't use snprintf */
-    char expectedBuffer[ TOPIC_BUFFER_SIZE + 1 ] = { 0 };
-    char thingName[ MAX_THING_NAME_LENGTH + 1 ] = { 0 };
-    size_t thingNameLength = 0U;
-    bool isMatch = true;
-
-    if ( topic == NULL || topicLength == 0 )
-    {
-        isMatch = false;
-    }
-
-    if ( isMatch )
-    {
-        mqttWrapper_getThingName( thingName, &thingNameLength );
-        snprintf( expectedBuffer,
-              TOPIC_BUFFER_SIZE,
-              "%s%s%s",
-              "$aws/things/",
-              thingName,
-              "/jobs/start-next/accepted" );
-        isMatch = ( uint32_t ) strnlen( expectedBuffer, TOPIC_BUFFER_SIZE ) ==
-              topicLength;
-        isMatch = isMatch && strncmp( expectedBuffer, topic, topicLength ) == 0;
-    }
+    return isThingnameTopicMatch(topic, topicLength, "/jobs/start-next/accepted", strlen("/jobs/start-next/accepted"));
+}
 
-    return isMatch;
+bool coreJobs_isJobUpdateStatus( const char * topic,
+                                const size_t topicLength,
+                                const char * jobId,
+                                const size_t jobIdLength,
+                                JobUpdateStatus_t expectedStatus )
+{
+    /* Max suffix size = max topic size - "$aws/<thingname>" prefix */
+    char suffixBuffer[ TOPIC_BUFFER_SIZE - MAX_THING_NAME_LENGTH - 4U] = { 0 };
+    char jobIdTerminated[ MAX_JOB_ID_LENGTH + 1 ] = { 0 };
+    char updateStatusString[ 10 ] = { 0 };
```

Comment:
Can we change this magic number with a Macro?

Reply:
Will do